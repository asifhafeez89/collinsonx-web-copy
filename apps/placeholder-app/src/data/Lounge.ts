enum AirportCode {
  BHD = 'BHD',
  BHX = 'BHX',
  BRS = 'BRS',
  EDI = 'EDI',
  HUY = 'HUY',
  INV = 'INV',
  LPL = 'LPL',
  MAN = 'MAN',
  NCL = 'NCL',
}

export type LoungeSchema = {
  LoungeCode: string;
  Partner: {
    IntegrationId: string;
    UID: string;
  };
  ServiceCentre: string;
  LoungeName: string;
  PPBOOperatorName: string;
  AirportCode: AirportCode;
  AirportName: string;
};

export const lounges: Array<LoungeSchema> = [
  {
    LoungeCode: 'BHD1',
    Partner: {
      IntegrationId: '001',
      UID: 'Swissport to Provide',
    },
    ServiceCentre: 'London',
    LoungeName: 'Aspire Lounge',
    PPBOOperatorName: 'Swissport GB Limited',
    AirportCode: AirportCode.BHD,
    AirportName: 'Belfast George Best City',
  },
  {
    LoungeCode: 'BIRM',
    Partner: {
      IntegrationId: '001',
      UID: 'Swissport to Provide',
    },
    ServiceCentre: 'London',
    LoungeName: 'Aspire Lounge',
    PPBOOperatorName: 'Swissport GB Limited',
    AirportCode: AirportCode.BHX,
    AirportName: 'Birmingham',
  },
  {
    LoungeCode: 'BHX4',
    Partner: {
      IntegrationId: '001',
      UID: 'Swissport to Provide',
    },
    ServiceCentre: 'London',
    LoungeName: 'Aspire Lounge (South)',
    PPBOOperatorName: 'Swissport GB Limited',
    AirportCode: AirportCode.BHX,
    AirportName: 'Birmingham',
  },
  {
    LoungeCode: 'BRS',
    Partner: {
      IntegrationId: '001',
      UID: 'Swissport to Provide',
    },
    ServiceCentre: 'London',
    LoungeName: 'Aspire Lounge',
    PPBOOperatorName: 'Swissport GB Limited',
    AirportCode: AirportCode.BRS,
    AirportName: 'Bristol International',
  },
  {
    LoungeCode: 'EDI2',
    Partner: {
      IntegrationId: '001',
      UID: 'Swissport to Provide',
    },
    ServiceCentre: 'London',
    LoungeName: 'Aspire Lounge (Gate 4)',
    PPBOOperatorName: 'Swissport GB Limited',
    AirportCode: AirportCode.EDI,
    AirportName: 'Edinburgh International',
  },
  {
    LoungeCode: 'HUY',
    Partner: {
      IntegrationId: '001',
      UID: 'Swissport to Provide',
    },
    ServiceCentre: 'London',
    LoungeName: 'Aspire Lounge',
    PPBOOperatorName: 'Swissport GB Limited',
    AirportCode: AirportCode.EDI,
    AirportName: 'Humberside Intl',
  },
  {
    LoungeCode: 'INV',
    Partner: {
      IntegrationId: '001',
      UID: 'Swissport to Provide',
    },
    ServiceCentre: 'London',
    LoungeName: 'Aspire Lounge',
    PPBOOperatorName: 'Swissport GB Limited',
    AirportCode: AirportCode.INV,
    AirportName: 'Inverness Dalcross',
  },
  {
    LoungeCode: 'LPL1',
    Partner: {
      IntegrationId: '001',
      UID: 'Swissport to Provide',
    },
    ServiceCentre: 'London',
    LoungeName: 'Aspire Lounge',
    PPBOOperatorName: 'Swissport GB Limited',
    AirportCode: AirportCode.LPL,
    AirportName: 'Liverpool John Lennon',
  },
  {
    LoungeCode: 'MAN6',
    Partner: {
      IntegrationId: '001',
      UID: 'Swissport to Provide',
    },
    ServiceCentre: 'London',
    LoungeName: 'Aspire Lounge',
    PPBOOperatorName: 'Swissport GB Limited',
    AirportCode: AirportCode.MAN,
    AirportName: 'Manchester International',
  },
  {
    LoungeCode: 'NCL',
    Partner: {
      IntegrationId: '001',
      UID: 'Swissport to Provide',
    },
    ServiceCentre: 'London',
    LoungeName: 'Aspire Lounge',
    PPBOOperatorName: 'Swissport GB Limited',
    AirportCode: AirportCode.NCL,
    AirportName: 'Newcastle International',
  },
];
