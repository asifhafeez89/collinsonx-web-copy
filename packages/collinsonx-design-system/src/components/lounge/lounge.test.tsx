import renderer from 'react-test-renderer';
import  Lounge  from '.';

describe('<Lounge />', () => {
    it('renders Lounge', () => {
        const tree = renderer
            .create(
                <Lounge
                    image="image.png"
                    airport="LA"
                    loungeName="Totally lounge"
                    openingTimes="10:00 - 12:00"
                    openDays="All day long"
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})