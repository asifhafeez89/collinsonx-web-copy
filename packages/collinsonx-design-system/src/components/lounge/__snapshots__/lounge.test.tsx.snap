// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Lounge /> renders Lounge 1`] = `
[
  <div
    className="mantine-Image-root mantine-15po0m8"
    style={
      {
        "width": "100%",
      }
    }
  >
    <figure
      className="mantine-11nhzn5 mantine-Image-figure"
    >
      <div
        className="mantine-qqmv3w mantine-Image-imageWrapper"
      >
        <img
          alt="LA"
          className="mantine-3y8yz3 mantine-Image-image"
          onError={[Function]}
          src="image.png"
          style={
            {
              "height": "auto",
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
      </div>
    </figure>
  </div>,
  <div
    className="mantine-Text-root mantine-16zcv3h"
  >
    Totally lounge
  </div>,
  <div
    className="mantine-SimpleGrid-root mantine-vt6ont"
  >
    <div
      className="mantine-1a78l0v"
    >
      <test-file-stub
        color="#000000"
      />
      <div
        className="mantine-2rwq4h"
      >
        <div
          className="mantine-Text-root mantine-15e5w66"
        >
          LA
        </div>
      </div>
    </div>
    <div
      className="mantine-1a78l0v"
    >
      <test-file-stub
        color="#000000"
      />
      <div
        className="mantine-2rwq4h"
      >
        <div
          className="mantine-Text-root mantine-15e5w66"
        >
          10:00 - 12:00
        </div>
      </div>
    </div>
  </div>,
]
`;
